pennylane/ops/qubit/hamiltonian.py

```diff
@@ -625,7 +625,7 @@ def compare(self, other):
         >>> ob1.compare(ob2)
         False
         """
-        if isinstance(other, Hamiltonian):
+        if isinstance(other, (Hamiltonian, qml.ops.LinearCombination)):
```

Comment:
Just a heads up that this was removed from `ham-tests`. Since that builds on top of this branch, I think it's okay to leave this as is.